core/src/mem.rs

```diff
@@ -0,0 +1,79 @@
+use crate::error::StoreError;
```

Comment:
As I understand this implementation is a few things - reference implementation for a Store, something built-in that rust-ipld can use if nothing else is implementing a store from the outside... is that correct?

Wondering if this would be better considered as a testing util and/or something in the examples folder? But maybe I'm missing the purpose of this.

Reply:
The above is mostly just a folder structure nitpick, but I think nailing that down will help future implementers / readers if it's categorized properly.

Reply:
I think it's useful as a reference implementation. Any store implementation should be "equivalent". If you want to operate on ipld in memory it does make sense to have such an implementation. But they are good points.

Reply:
Agree it's a great reference

Reply:
Well it would be if it were correct, but I think I'll leave that for a follow up.